common/service.go

```diff
@@ -65,18 +64,23 @@ Typical usage:
 		// stop subroutines, etc.
 	}
 */
+
+// BaseService implements Service.
 type BaseService struct {
-	Logger  log.Logger
+	Logger log.Logger
+	Quit   chan struct{}
+
 	name    string
 	started uint32 // atomic
 	stopped uint32 // atomic
-	Quit    chan struct{}
 
 	// The "subclass" of BaseService
 	impl Service
 }
 
-func NewBaseService(logger log.Logger, name string, impl Service) *BaseService {
+// NewBaseService returns a pointer to a newly instantiated BaseService object.
+// logger can be nil and will then default to a NewNopLogger.
+func NewBaseService(name string, impl Service, logger log.Logger) *BaseService {
```

Comment:
unless there's really good reason we shouldn't go changing the order of parameters ... its just a big nuisance downstream

Reply:
Yeah, I was thinking about this one too.

Actually I came across a situation where it makes sense to keep it in front. I'll change it to the front.

I just reordered it for style :-)